exonum/src/blockchain/mod.rs

```diff
@@ -108,40 +108,43 @@ impl StaticServiceDispatcher {
     }
 
     // TODO clarify if we need this method
-    pub fn services(&self) -> Arc<HashMap<u16, Box<dyn Service>>>
-    {
+    pub fn services(&self) -> Arc<HashMap<u16, Box<dyn Service>>> {
```

Comment:
I don't know how it is actual for now, but why don't you wrap `HashMap` with `RwLock` ? Is it redundant ?

Reply:
I just copy-pasted it from the [old version](https://github.com/exonum/exonum/blob/master/exonum/src/blockchain/mod.rs#L80) :) 
Anyway, this dispatcher will be removed soon. It was created mostly by copy-pasting just to make new dispatched integration easier in the future.